iFixit/react-commerce
# Filters: Reset page to 0 on filter change

Closes https://github.com/iFixit/react-commerce/issues/1005, https://github.com/iFixit/react-commerce/issues/697

Issue (description from #1005)
--
When on a device page (like https://www.ifixit.com/Parts/iPhone_6s?p=2), and you click an item type (like Case Components), the item type is applied but the other facets remain, and consequently sometimes there are no results.

QA
--

1. Go to a device page with multiple pages
2. Navigate to the second page
3. Apply an `Item Type` filter
4. Ensure that the newly filtered page is no longer on the second page


cr_req 2
